zeus/run/dataloader.py

```diff
@@ -134,22 +154,83 @@ def __init__(
             batch_size: Batch size to use for training.
             max_epochs: Maximum number of epochs to train. **Specify this parameter only
                 to the train data loader.**
+            distributed: Distributed strategy to use for training. If training with single GPU,
+                this value should be None; if training using data parallel with multi-GPU on
+                a single node, this value should be "dp".
+
+        Raises:
+            RuntimeError: `torch.distributed` package is not available.
+            RuntimeError: The default process group is not initialized. Make sure to call
+                `torch.distributed.init_process_group` to initialize the default process
+                group before doing a multiprocessing distributed training.
+            ValueError: `self.sampler` is not an instance of DistributedSampler. An instance of
+                DistributedSampler will shuffle and distribute data among GPUs, so it is required
+                for data paralle training.
+            ValueError: DistributedSampler passed in self.sampler is inconsistent with the default
+                process group. Currently, we assume that all the GPUs in the node will be used for
+                training. In this case, the DistributedSampler should have
+                `sampler.num_replicas == torch.distributed.get_world_size()`
+                and `sampler.rank == torch.distributed.get_rank()`.
         """
         # Sanity checks.
         self.batch_size = batch_size
         self.split = "train" if max_epochs != -1 else "eval"
         self.max_epochs = max_epochs
-        self.log_prefix = f"\n[ZeusDataLoader({self.split})]"
+        self.log_prefix = f"[ZeusDataLoader({self.split})]"
+
+        # Initialize the DataLoader.
+        super().__init__(*args, batch_size=batch_size, **kwargs)
+
+        # World size and rank for distributed training.
+        # Set default value for single-GPU.
+        self.world_size = 1
+        self.rank = 0
+        # Check whether we are doing a distributed training.
+        # Pass in world size and rank.
+
+        self.distributed = distributed
+
+        if self.distributed == "dp":
+            # Check if the distributed package is available.
+            if not dist.is_available():
+                raise RuntimeError("Requires distributed package to be available.")
+            # Check if the process group is initialized.
+            if not dist.is_initialized():
+                raise RuntimeError(
+                    "Default process group has not been initialized,"
+                    " please make sure to call init_process_group."
+                )
+            # Check if `self.sampler` is an instance of DistributedSampler.
+            if not isinstance(getattr(self, "sampler", None), DistributedSampler):
```

Comment:
Actually, I think this has to be in the class docstring & documentation, because this is part of the contract of `ZeusDataLoader`. We're basically asking the user to always use a `DistributedSampler` with `ZeusDataLoader`.